Dockerfile.minimal

```diff
@@ -0,0 +1,150 @@
+COPY libdevice_fix.sh /tmp/libdevice_fix.sh
+# ==================================================================
+# Miniconda
+# ------------------------------------------------------------------
+RUN wget --quiet https://repo.anaconda.com/miniconda/Miniconda3-py311_23.5.2-0-Linux-x86_64.sh -O ~/miniconda.sh && \
+    /bin/bash ~/miniconda.sh -b -p /opt/conda && \
+    rm ~/miniconda.sh && \
+    ln -s /opt/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh && \
+    echo ". /opt/conda/etc/profile.d/conda.sh" >> ~/.bashrc && \
+    echo "conda activate base" >> ~/.bashrc && \
+    . /opt/conda/etc/profile.d/conda.sh && \
+    conda activate base && \
+    . /tmp/libdevice_fix.sh && \
+# ==================================================================
+# Python
+# ------------------------------------------------------------------
+    PIP_INSTALL="python -m pip --no-cache-dir install --upgrade" && \
+    $PIP_INSTALL pip pipx && \
+    python3 -m pipx ensurepath && \
+    $PIP_INSTALL -r /tmp/requirements/python.txt && \
+    rm -r /tmp/requirements
+# ==================================================================
+# OOM guard
+# Adds a script to tune oom_killer behavior and puts it into the crontab
+# ==================================================================
+COPY files/usr/local/sbin/oom_guard.sh /usr/local/sbin/oom_guard.sh
+RUN crontab -l 2>/dev/null | { cat; echo '* * * * * /usr/local/sbin/oom_guard.sh'; } | crontab
+
+# ==================================================================
+# Set up SSH for remote debug
+# ------------------------------------------------------------------
+
+# Setup environment for ssh session
+RUN apt-get install -y --no-install-recommends openssh-server && \
+    echo "export PATH=/root/.local/bin:$PATH" >> /etc/profile && \
+    echo "export LANG=$LANG" >> /etc/profile && \
+    echo "export LANGUAGE=$LANGUAGE" >> /etc/profile && \
+    echo "export LC_ALL=$LC_ALL" >> /etc/profile && \
+    echo "export PYTHONIOENCODING=$PYTHONIOENCODING" >> /etc/profile && \
+    . /etc/profile && \
+    apt-get clean && \
+    apt-get autoremove && \
+    rm -rf /var/lib/apt/lists/* /tmp/* ~/*
+
+# Create folder for openssh fifos
+RUN mkdir -p /var/run/sshd
+
+# Disable password for root
+RUN sed -i -re 's/^root:[^:]+:/root::/' /etc/shadow
+RUN sed -i -re 's/^root:.*$/root::0:0:System Administrator:\/root:\/bin\/bash/' /etc/passwd
+
+# Permit root login over ssh
+RUN echo "Subsystem    sftp    /usr/lib/sftp-server \n\
+PasswordAuthentication yes\n\
+ChallengeResponseAuthentication yes\n\
+PermitRootLogin yes \n\
+PermitEmptyPasswords yes\n" > /etc/ssh/sshd_config
+
+# ssh port
+EXPOSE 22
+
+# ==================================================================
+# Neu.ro and other isolated via pipx Python packages
+# ------------------------------------------------------------------
+COPY requirements/pipx.minimal.txt /tmp/requirements/pipx.txt
```

Comment:
minor: we could put the requirements file name used here to args and have one dockerfile to support, but its up to you. re vscode and the notebook, don't think it will take much space

Reply:
I'd say we go with the minimal version as it is now, and in case there is a demand for an IDE in the minimal image we can return it

Reply:
my point was mostly re single dockerfile, but this is minor, we could have it as is :)